src/components/eventsPageComponents/EventList.jsx

```diff
@@ -1,30 +1,77 @@
 import { onSnapshot } from "firebase/firestore";
+import { useRouter } from "next/navigation";
+import { useTranslation } from "next-i18next";
 import React, { useEffect, useState } from "react";
 
+import { useUser } from "@/context/UserContext";
+
+import DateFilter from "./DateFilter";
 import EventCard from "./EventCard";
 import { eventsCollection } from "../../lib/firebase/controller";
 
 function EventList() {
     const [events, setEvents] = useState([]);
-
+    const [filteredEvents, setFilteredEvents] = useState([]);
+    const { user } = useUser();
+    const router = useRouter();
+    // sets the routing according to the user's state
+    const handleUser = () => {
+        user
+            ? router.push("/createEvent")
+            : router.push("/authentication/signUp");
+    };
+    // var to handle translatons
+    const { t } = useTranslation();
+    const onClick = (selectedDate) => {
```

Comment:
it's better to give functions meaningfull name, i.e handleDateFiltering

Reply:
sorry for the confusion but this is for the translation variable